### PR TITLE
Chore: Rewrite test helpers from GoConvey to stdlib

### PR DESCRIPTION
**Pull request by ceh**
**What this PR does / why we need it**:

Rewrite two test helpers, `createUser` and `insertTestDashboard`, in the `sqlstore` package to not require a GoConvey context, to make rewriting GoConvey tests to stdlib and testify easier.

**Special notes for your reviewer**:

Part of #22844.

What do you think @aknuds1, @papagian?

### CODE REVIEW

File: pkg/services/sqlstore/dashboard_test.go [hunk 1]
```diff
@@ -442,7 +449,7 @@ func TestDashboard_SortingOptions(t *testing.T) {
 	})
 }
 
-func insertTestDashboard(title string, orgId int64, folderId int64, isFolder bool, tags ...interface{}) *models.Dashboard {
+func insertTestDashboard(title string, orgId int64, folderId int64, isFolder bool, tags ...interface{}) (*models.Dashboard, error) {
```
Comment (aknuds1) [inline pkg/services/sqlstore/dashboard_test.go:452]: I think with test utility functions, it's better to make them take a `*testing.T` first argument and use `require.NoError(t, err)` etc. Also stick a `t.Helper()` call at the beginning of the function. You're in the context of a test, so you might as well use the `testing.T` facilities.

**Review by aknuds1 — COMMENTED** [review]
I think that, instead of making test utilities return an error, it's better to adjust them to take a `*testing.T` first argument and use `require.NoError(t, err)` etc, as if you were right inside a test function. I don't see any reason not to use `testing.T` facilities within test helpers. WDYT?

**Comment by ceh** [comment]
I updated the PR @aknuds1, please take another look.

File: pkg/services/sqlstore/dashboard_test.go [hunk 2]
```diff
@@ -481,19 +482,21 @@ func insertTestDashboardForPlugin(title string, orgId int64, folderId int64, isF
...
 	err := CreateUser(context.Background(), &currentUserCmd)
-	So(err, ShouldBeNil)
+	require.NoError(t, err)
 
 	q1 := models.GetUserOrgListQuery{UserId: currentUserCmd.Result.Id}
 	err = GetUserOrgList(&q1)
-	So(err, ShouldBeNil)
-	So(q1.Result[0].Role, ShouldEqual, role)
+	require.NoError(t, err)
+	require.Equal(t, q1.Result[0].Role, models.RoleType(role))
```
Comment (aknuds1) [inline pkg/services/sqlstore/dashboard_test.go:499]: The expectation comes before the actual value in `require.Equal`:
```suggestion
	require.Equal(t, models.RoleType(role), q1.Result[0].Role)
```

**Review by aknuds1 — APPROVED** [review]
LGTM

**Comment by aknuds1** [comment]
Do you mind applying my suggestion before I merge?

File: pkg/services/sqlstore/dashboard_test.go (hunk 2 above)
Comment (ceh) [reply pkg/services/sqlstore/dashboard_test.go]: Ah, yes. Fixed. Thanks!

**Comment by aknuds1** [comment]
Thanks for the contribution!

**Pull request merged**